File: src/v5/Customer.java
package v5;

import java.util.Random;

class Customer extends Person {
    private  String subscriptionType;
    private  int subscriptionQuantity;
    private final String status;
    private double balance;

    public Customer(String name, String nickname,String gender, String phoneNumber,String address, String email,String subscriptionType, int subscriptionQuantity) {
        super( name, nickname,gender, phoneNumber, address, email);
        this.subscriptionQuantity=subscriptionQuantity;
        this.subscriptionType=subscriptionType;
        Random random = new Random();
        boolean randomBoolean = random.nextBoolean();
        status = randomBoolean?"Payed":"Not Payed";
    }

    // Getters and Setters for Customer properties

    public double getBalance() {
        return balance;
    }

    public void setBalance(double balance) {
        this.balance = balance;
    }

    public String getSubscriptionType() {
        return subscriptionType;
    }

    public String getStatus() {
        return status;
    }

    public void setSubscriptionType(String subscriptionType) {
        this.subscriptionType = subscriptionType;
    }

    public int getSubscriptionQuantity() {
        return subscriptionQuantity;
    }

    public void setSubscriptionQuantity(int subscriptionQuantity) {
        this.subscriptionQuantity = subscriptionQuantity;
    }

    @Override
    public String getGender() {
        return super.getGender();
    }

    @Override
    public void setGender(String gender) {
        super.setGender(gender);
    }

    @Override
    public int getId() {
        return super.getId();
    }

    @Override
    public String getAddress() {
        return super.getAddress();
    }


    @Override
    public String getEmail() {
        return super.getEmail();
    }

    @Override
    public String getName() {
        return super.getName();
    }

    @Override
    public String getNickname() {
        return super.getNickname();
    }

    @Override
    public String getPhoneNumber() {
        return super.getPhoneNumber();
    }

    @Override
    public void setAddress(String address) {
        super.setAddress(address);
    }

    @Override
    public void setEmail(String email) {
        super.setEmail(email);
    }

    @Override
    public void setName(String name) {
        super.setName(name);
    }

    @Override
    public void setNickname(String nickname) {
        super.setNickname(nickname);
    }

    @Override
    public void setPhoneNumber(String phoneNumber) {
        super.setPhoneNumber(phoneNumber);
    }
}

File: pythonCodeConversion/electricbill/models/customer.py
import random

from models.person import Person


class Customer(Person):
    def __init__(self, name, nickname, gender, phone_number, address, email, subscription_type, ampere, balance):
        super().__init__(name, nickname, gender, phone_number, address, email)
        self.subscriptionQuantity = ampere
        self.subscriptionType = subscription_type
        self.balance = balance
        random_boolean = random.choice([True, False])
        self.status = "Payed" if random_boolean else "Not Payed"

    def get_balance(self):
        return self.balance

    def get_subscription_type(self):
        return self.subscriptionType

    def get_status(self):
        return self.status

    def get_subscription_quantity(self):
        return self.subscriptionQuantity

    # Override methods from the Person class
    def get_gender(self):
        return super().get_gender()

    def set_gender(self, gender):
        super().set_gender(gender)

    def get_id(self):
        return super().get_id()

    def get_address(self):
        return super().get_address()

    def get_email(self):
        return super().get_email()

    def get_name(self):
        return super().get_name()

    def get_nickname(self):
        return super().get_nickname()

    def get_phone_number(self):
        return super().get_phone_number()

    def set_address(self, address):
        super().set_address(address)

    def set_email(self, email):
        super().set_email(email)

    def set_name(self, name):
        super().set_name(name)

    def set_nickname(self, nickname):
        super().set_nickname(nickname)

    def set_phone_number(self, phone_number):
        super().set_phone_number(phone_number)
